src/storage/abstractions.rs

```diff
@@ -0,0 +1,44 @@
+use crate::Channel;
+
+/// The mechanism used to manage the storage of the inner state of a channel of `T`.
+#[expect(private_bounds, reason = "sealed trait with private API surface")]
```

Comment:
I observe this attribute is failing one of the automated builds that uses Rust 1.65 toolchain, as the lint and the way I suppressed it are all from a too-new Rust version. What is the correctly pattern to apply here to suppress this linter warning in context of this repo's practices? I am not very familiar with how to best build using both old and new toolchains at the same time (except in dependency scenarios where lints are suppressed anyway).

Reply:
Okay, relaxing to an `allow()` got rid of some of the warnings but I see that 1.65 does not allow this pattern at all. Will leave the question of what to do with this open for now, pending design discussion.

Reply:
One option to avoid the sealing being an issue on older Rust is to just open up the pattern to non-sealed, so anyone can make an `impl Storage`.